charts/ingress-apisix/templates/deployment.yaml

```diff
@@ -0,0 +1,92 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "ingress-apisix.fullname" . }}
+  labels:
+    {{- include "ingress-apisix.labels" . | nindent 4 }}
+spec:
+  {{- if not .Values.ingressController.autoscaling.enabled }}
+  replicas: {{ .Values.ingressController.replicaCount }}
+  {{- end }}
+  selector:
+    matchLabels:
+      {{- include "ingress-apisix.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+      {{- with .Values.ingressController.podAnnotations }}
+      annotations:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      labels:
+        {{- include "ingress-apisix.selectorLabels" . | nindent 8 }}
+    spec:
+      {{- with .Values.ingressController.imagePullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      serviceAccountName: {{ .Values.ingressController.serviceAccount }}
+      securityContext:
+        {{- toYaml .Values.ingressController.podSecurityContext | nindent 8 }}
+      volumes:
+        - name: configuration
+          configMap:
+            name: {{ .Values.ingressController.configmap }}
+            items:
+              - key: config.yaml
+                path: config.yaml
+      containers:
+        - name: {{ .Chart.Name }}
+          command:
+            - /ingress-apisix/apisix-ingress-controller
+            - ingress
+            - --config-path
+            - /ingress-apisix/conf/config.yaml
+          securityContext:
+            {{- toYaml .Values.ingressController.securityContext | nindent 12 }}
+          image: "{{ .Values.ingressController.image.repository }}:{{ .Values.ingressController.image.tag | default .Chart.AppVersion }}"
+          imagePullPolicy: {{ .Values.ingressController.image.pullPolicy }}
+          ports:
+            - name: http
+              containerPort: {{ (.Values.ingressController.config.httpListen | split ":")._1 }}
+              protocol: TCP
+          livenessProbe:
+            httpGet:
+              path: /healthz
+              port: http
```

Comment:
Here must be a port .

Reply:
port in httpGet action can be number or string.

> Name or number of the port to access on the container. Number must be in the range 1 to 65535. Name must be an IANA_SVC_NAME.

Reply:
See https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.19/#httpgetaction-v1-core